app/controllers/criteria_controller.rb

```diff
@@ -0,0 +1,17 @@
+class CriteriaController < ApplicationController
+
+  # This method handles the drag/drop criteria sorting
+  def update_positions
+    @assignment = Assignment.find(params[:assignment_id])
+    @criteria = @assignment.get_criteria
+    position = 0
+
+    params[:criterion].each do |id|
```

Comment:
By the way, thank you for taking the time to review my pull request during the weekend.